lib/snippets.coffee

```diff
@@ -128,11 +128,55 @@ module.exports =
         @add(userSnippetsPath, result)
 
   loadPackageSnippets: (callback) ->
+    # Sort it *before* mapping it
     packages = atom.packages.getLoadedPackages()
-    snippetsDirPaths = (path.join(pack.path, 'snippets') for pack in packages).sort (a, b) ->
-      if /\/app\.asar\/node_modules\//.test(a) then -1 else 1
-    async.map snippetsDirPaths, @loadSnippetsDirectory.bind(this), (error, results) ->
-      callback(_.extend({}, results...))
+      .sort (pack, b) ->
+        if /\/app\.asar\/node_modules\//.test(pack.path) then -1 else 1
+
+    # So the derived array has the same order
+    snippetsDirPaths = (path.join(pack.path, 'snippets') for pack in packages)
+    async.map snippetsDirPaths, @loadSnippetsDirectory.bind(this), (error, results) =>
```

Comment:
💭 I wonder why we're using `async.map` here?  Is the native `Array.map` not performant enough to do synchronously?

Just some thoughts, no need to address them in this PR.